lib/ansible/module_utils/rds.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+import botocore
+
+from ansible.module_utils.ec2 import camel_dict_to_snake_dict
+
+
+def get_db_instance(conn, instancename):
+    try:
+        response = conn.describe_db_instances(DBInstanceIdentifier=instancename)
+        instance = RDSDBInstance(response['DBInstances'][0])
+        return instance
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBInstanceNotFound':
+            return None
+        else:
+            raise
+
+
+def get_db_snapshot(conn, snapshotid):
+    try:
+        response = conn.describe_db_snapshots(DBSnapshotIdentifier=snapshotid)
+        snapshot = RDSSnapshot(response['DBSnapshots'][0])
+        return snapshot
+    except botocore.exceptions.ClientError as e:
+        if e.response['Error']['Code'] == 'DBSnapshotNotFound':
+            return None
+        else:
+            raise
+
+
+class RDSDBInstance(object):
```

Comment:
This feels to me like it would be better as a function that would take a dbinstance and convert it to the facts dict, then a separate one to diff dictionaries rather than a class.